packages/test/test-end-to-end-tests/src/test/gc/gcAttachmentBlobs.spec.ts

```diff
@@ -254,48 +272,50 @@ describeNoCompat("Garbage collection of blobs", (getTestObjectProvider) => {
+            // the blobs uploaded in detached mode should map to the same storageId.
+            const localHandle3 = await defaultDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
 
-            // Store the localId1 and localId2 handles. This would make the localId nodes and storageId node referenced.
+            // Validate that storing the localId handles makes both the localId and storageId nodes as referenced since
+            // localId is simply an alias to the storageId.
             defaultDataStore._root.set("local1", localHandle1);
             defaultDataStore._root.set("local2", localHandle2);
+            defaultDataStore._root.set("local3", localHandle3);
             const s1 = await summarizeAndGetUnreferencedNodeStates(summarizerRuntime);
-            assert(s1.get(localHandle1.absolutePath) === "referenced", "local id 1 blob should be referenced");
-            assert(s1.get(localHandle2.absolutePath) === "referenced", "local id 2 blob should be referenced");
-            assert(s1.get(storageHandle.absolutePath) === "referenced", "storage id blob should be referenced (1)");
-
-            // Delete the localId1 handle. This would make localId1 node unreferenced.
+            assert.strictEqual(s1.size, 4, "There should be 4 blob entries in GC data");
+            const storageId = getStorageIdFromReferenceMap(
+                s1,
+                [localHandle1.absolutePath, localHandle2.absolutePath, localHandle3.absolutePath],
+            );
+            assert(s1.get(localHandle1.absolutePath) === "referenced", "local id 1 blob should be referenced (1)");
+            assert(s1.get(localHandle2.absolutePath) === "referenced", "local id 2 blob should be referenced (1)");
+            assert(s1.get(localHandle3.absolutePath) === "referenced", "local id 3 blob should be referenced (1)");
+            assert(s1.get(storageId) === "referenced", "storage id blob should be referenced (1)");
+
+            // Delete the localId handles. This would make localId1 node unreferenced.
             defaultDataStore._root.delete("local1");
+            defaultDataStore._root.delete("local2");
+            defaultDataStore._root.delete("local3");
             const s2 = await summarizeAndGetUnreferencedNodeStates(summarizerRuntime);
             assert(s2.get(localHandle1.absolutePath) === "unreferenced", "local id 1 blob should be unreferenced");
-            assert(s2.get(localHandle2.absolutePath) === "referenced", "local id 2 blob should be referenced");
-            assert(s2.get(storageHandle.absolutePath) === "referenced", "storage id blob should be referenced (2)");
+            assert(s2.get(localHandle2.absolutePath) === "unreferenced", "local id 2 blob should be referenced (2)");
+            assert(s2.get(localHandle3.absolutePath) === "unreferenced", "local id 3 blob should be referenced (2)");
+            assert(s2.get(storageId) === "unreferenced", "storage id blob should be referenced (2)");
```

Comment:
Nit: Should be unreferenced instead of referenced

Reply:
Right. There are other instances where this is wrong, I will fix them all.